components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/internal/util/ActionManagementAuditLogger.java

```diff
@@ -70,6 +71,39 @@ public void printAuditLog(Operation operation, String actionType, String actionI
         buildAuditLog(operation, data);
     }
 
+    /**
+     * Print action audit log related to the operation by the action type and action ID with updated time.
+     *
+     * @param operation  Operation associated with the state change.
+     * @param actionType Type of the action to be logged.
+     * @param actionId   ID of the action to be logged.
+     * @param updatedAt  Time of the update.
+     */
+    public void printAuditLog(Operation operation, String actionType, String actionId, Timestamp updatedAt) {
+        JSONObject data = createAuditLogEntry(actionType, actionId);
+        data.put(LogConstants.UPDATED_AT_FIELD, updatedAt != null ? updatedAt : JSONObject.NULL);
+        buildAuditLog(operation, data);
+    }
+
+    /**
+     * Print action audit log related to the operation by the action DTO with updated time.
+     *
+     * @param operation         Operation associated with the state change.
+     * @param updatingActionDTO Action object to be logged.
+     * @param updatedAt         Time of the update.
+     */
+    public void printAuditLog(Operation operation, ActionDTO updatingActionDTO, Timestamp updatedAt) {
+        JSONObject data = null;
+        try {
+            data = createAuditLogEntry(updatingActionDTO);
+        } catch (ActionMgtException e) {
+            data = createAuditLogEntry(String.valueOf(updatingActionDTO.getType()), updatingActionDTO.getId());
+        }
```

Comment:
Why are we doing like this ?
Let's catch the exception and add a warn log and proceed.